enhancements/sdk-migrate-bundle.md

```diff
@@ -0,0 +1,379 @@
+
+As an operator developer, I would like to migrate my operator which uses
+the packagemanifest format to the new bundle format.
+
+#### Story 2
+
+As an catalog admin for my operators, I would like to be able to migrate a
+collection of packagemanifests to bundles.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The Operator SDK currently supports generating & validating
+packagemanifests & bundles. This enhancement will add a new command, `migrate
+bundle` to the CLI. This new command will migrate an existing packagemanifest to
+a bundle.
+
+More information about the different format can be found at the URLs below:
+
+- Operator bundles: https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md
+- Package manifests format: https://github.com/operator-framework/operator-registry/tree/v1.5.3/#manifest-format
+
+The Operator SDK will traverse the given packagemanifest directory and for each
+of the versions create a new bundle.
+
+#### Generating bundle on disk
+
+In order to generate bundles to disk, we will refactor the `generate bundle`
+logic to be reusable in generating the bundles.
+
+===================== REWRITE BELOW =======================
+The `GenerateFunc` takes in the following arguments:
+
+- directory string - local directory where the manifests and metadata are
+  located
+- outputDir string - where manifests and metadata are copied
+- packageName string - the name of the package that bundle image belongs to
+- channels string - the list of channels that bundle image belongs to
+- channelDefault string - the default channel for the bundle image
+
+In order to use this function, we will obtain the required parameters from the
+listed *source*:
+
+| parameter | source |
+| --------- | ------ |
+| directory | packagemanifest directory |
+| outputDir | default `bundle` dir or flag `--output-dir` |
+| packageName | packageName from `*.package.yaml` |
+| channels | Any channel where `channels[].currentCSV == packagedCSV.Name` from `*.package.yaml` |
+| channelDefault | defaultChannel from `*.package.yaml` or `stable` |
+===================== REWRITE ABOVE =======================
+
+#### Creating bundle image
+
+If we choose to build bundle images, we will likely need a sane default build
+command but also offer an override.
+
+This would add 2 more flags to the CLI:
+
+- `[--image-base]` optional flag that indicates the base container name for
+  the bundles; e.g. `quay.io/example/memcached-operator-bundle`
+- `[--build-cmd]` optional flag that indicates the fully qualified build command;
+
+These flags would output a bundle image for each bundle created. We would use
+the `<dirname>` for the tags.
+
+The `--image-base` needs to be valid docker image name without the tag. We will
+use the bundle directory for the imagetag.
+
+Since we can't always anticipate what folks want to do with their builds, we'll
+have an excape hatch flag, `--build-cmd`, which will take in the full build
```

Comment:
in other places, i think we are calling container-tools. could we check and try to have a standard?
It makes the user experience easier. e.g you know that with command A you are able to choose if you will use docker or podman with the flag X then when you are using another command you will try to find out the same flag intuitively first.